vortex-dtype/src/composite.rs

```diff
@@ -0,0 +1,50 @@
+use std::fmt::{Display, Formatter};
+
+use linkme::distributed_slice;
+use vortex_error::{vortex_err, VortexError};
+
+#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd, Hash)]
+#[cfg_attr(feature = "serde", derive(::serde::Serialize))]
+pub struct CompositeID(&'static str);
+
+impl CompositeID {
```

Comment:
Given pr title I am surprised to see these changes here. Can we change the pr title or make separate pr?